interface/reports/reports_controllers/AppointmentsController.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/*
+ * These functions are common functions used in the Appointments reports. They have pulled out 
+ * and placed in this file. This is done to prepare the for building a
+ * report generator.
+ *
+ * Copyright (C) 2018 Tigpezeghe Rodrige <tigrodrige@gmail.com>
+ * Copyright (C) 2015-2017 Terry Hill <teryhill@librehealth.io>
+ * Copyright (C) 2005-2010 Rod Roark <rod@sunsetsystems.com>
+ *
+ * LICENSE: This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
+ * See the Mozilla Public License for more details.
+ * If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * @package LibreHealth EHR
+ * @author Tigpezeghe Rodrige <tigrodrige@gmail.com>
+ * @link http://librehealth.io
+ *
+ * Please help the overall project by sending changes you make to the author and to the LibreHealth EHR community.
+ *
+ */
+
+$fake_register_globals=false;
+$sanitize_all_escapes=true;
+
+require_once("../globals.php");
+require_once("../../library/patient.inc");
+require_once("../../library/report_functions.php");
+require_once("$srcdir/formatting.inc.php");
+require_once "$srcdir/options.inc.php";
+require_once "$srcdir/formdata.inc.php";
+require_once "$srcdir/appointments.inc.php";
+require_once "$srcdir/clinical_rules.php";
+$DateFormat = DateFormatRead();
+$DateLocale = getLocaleCodeForDisplayLanguage($GLOBALS['language_default']);
+
+# Clear the pidList session whenever load this page.
+# This session will hold array of patients that are listed in this 
+# report, which is then used by the 'Superbills' and 'Address Labels'
+# features on this report.
+unset($_SESSION['pidList']);
+
+$alertmsg = ''; // not used yet but maybe later
+
+if ($patient && ! $_POST['form_from_date']) {
+    // If a specific patient, default to 2 years ago.
+    $tmp = date('Y') - 2;
+    $from_date = date("$tmp-m-d");
+} else {
+    $from_date = fixDate($_POST['form_from_date'], date('Y-m-d'));
```

Comment:
begin_date and end_date are my favorites, but I still recommend that when posting a value to a variable, you use the same base name for continuity (unless you need a second variable).  As in "$form_from_date".  This helps track the stuff, and you always know that the php variable is one that can only have a value from the HTML.

Reply:
feel free to ignore this foible of mine :)